backend/src/controller/book.ts

```diff
@@ -31,7 +33,7 @@ const createBook = async (req: Request, res: Response) => {
 //Get all books
 const getAllBooks = async (req: Request, res: Response) => {
   try {
-    const result = await bookService(dbConnection).getAll();
+    const result = await genericService(dbConnection, "books", Book).getAll();
```

Comment:
Aren't you missing the Diamond operator to specify T? I would expect you get typescript errors otherwise.

`genericService<Book>(dbConnection, "books", Book).getAll();`